scripts/benchmarks/benchmark.py

```diff
@@ -690,6 +701,15 @@ def prepare_arguments(args, benchmark_dataset: BenchmarkDataset):
                     else constants.get(HF_ARG_RESPONSE_TEMPLATE)
                 ),
             )
+            # Check to remove all template arguments if dataset is pretokenized at this stage
```

Comment:
see comment above. I think if you keep putting these kind of logic inside the benchmark script it will never end, because we cannot control what kind of flags and checks they put in `tuning.sft_trainer`

Reply:
Lets instead update the `prepare_dataset` logic to accept a default `response_template`

```
    if self.response_template:
            if response_template is not None:
                warning.warn("overiding template ")
            response_template = self.response_template
```